src/topojsonm/topojsonm.ml

```diff
@@ -171,16 +171,17 @@ let map_object f src dst =
   in
   let rec go () =
     match Jsonm.decode decoder with
-    | `Lexeme (`Name "object" as t) ->
+    | `Lexeme (`Name "objects" as t) ->
         (match Jsonm.decode decoder with
-        | `Lexeme (`Name "example" as ex) -> (
+        | `Os -> (
+        let b = Jsonm.decode decoder in
             match Topo.of_json @@ decode_single_object decoder with
             | Error (`Msg m) -> raise (Abort (`Unexpected m))
             | Ok v -> (
                 match Topo.topojson v with
                 | Geometry g ->
-                    let g' = f g in
-                    enc (`Lexeme ex);
+                    let g' = f b g in
```

Comment:
This isn't quite right, remember as well we want to pass the name and geometry to `f`. Here's a function that you might find useful. 

```ocaml
let rec loop_through_objects decoder = match Jsonm.decode decoder with
    | `Lexeme `Oe -> ignore (enc (`Lexeme `Oe)); ()
    | `Lexeme (`Name geometry_name) -> (
      let geometry_json = decode_single_object decoder in
      match Topo.Geometry.of_json geometry_json with
      | Error (`Msg m) -> failwith m
      | Ok g ->
        let new_name, new_geometry = f (geometry_name, g) in
        enc (`Lexeme (`Name new_name));
        encode_value encoder (Topo.Geometry.to_json new_geometry);
        loop_through_objects decoder
    )
    | _ -> failwith "Unexpected lexeme"
  in
```

This function allows you to loop through the objects, apply the function and re-encode them. That combined with all of the work in this PR should get you pretty close to a working solution.